Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java

```diff
@@ -304,7 +298,21 @@ public void run() {
             }
         }
 
+        class DelayMotdTask implements Runnable {
+            @Override
+            public void run() {
+                final User user = ess.getUser(player);
+
+                if (input != null && user.isAuthorized("essentials.motd")) {
+                    final IText output = new KeywordReplacer(input, user.getSource(), ess);
+                    final TextPager pager = new TextPager(output, true);
+                    pager.showPage("1", null, "motd", user.getSource());
+                }
+            }
+        }
+
         ess.scheduleSyncDelayedTask(new DelayJoinTask());
+        ess.scheduleSyncDelayedTask(new DelayMotdTask(), ess.getSettings().getMotdDelay());
```

Comment:
Check if `getMotdDelay` is 0, and immediately call `run`. Otherwise, this will not truly be 0ms. If the user cares enough for the quickest delay they can set delay to 1 in config.